ios/RCTWebRTC/SerializeUtils.m

```diff
@@ -193,21 +193,65 @@ + (NSString *) serializeDirection: (RTCRtpTransceiverDirection) direction {
 }
 
 + (RTCRtpTransceiverDirection) parseDirection: (NSString *) direction {
-    if ([direction  isEqual: @"inactive"]) {
+    if ([direction isEqual: @"inactive"]) {
         return RTCRtpTransceiverDirectionInactive;
-    } else if ([direction  isEqual: @"recvonly"]) {
+    } else if ([direction isEqual: @"recvonly"]) {
         return RTCRtpTransceiverDirectionRecvOnly;
-    } else if ([direction  isEqual: @"sendonly"]) {
+    } else if ([direction isEqual: @"sendonly"]) {
         return RTCRtpTransceiverDirectionSendOnly;
-    } else if ([direction  isEqual: @"sendrecv"]) {
+    } else if ([direction isEqual: @"sendrecv"]) {
         return RTCRtpTransceiverDirectionSendRecv;
-    } else if ([direction  isEqual: @"stopped"]) {
+    } else if ([direction isEqual: @"stopped"]) {
         return RTCRtpTransceiverDirectionStopped;
     }
     
     return RTCRtpTransceiverDirectionInactive;
 }
 
++ (RTCRtpEncodingParameters *) parseEncoding: (NSDictionary *) params {
+    RTCRtpEncodingParameters *encoding = [RTCRtpEncodingParameters new];
+
+    if (params[@"rid"] != nil) {
+      [encoding setRid: params[@"rid"]];
+    }
+    if (params[@"active"] != nil) {
+      [encoding setIsActive: ((NSNumber *) params[@"active"]).boolValue];
+    }
+    if (params[@"maxBitrate"] != nil) {
+      [encoding setMaxBitrateBps: (NSNumber *) params[@"maxBitrate"]];
+    }
+    if (params[@"maxFramerate"] != nil) {
+      [encoding setMaxFramerate: (NSNumber *) params[@"maxFramerate"]];
+    }
+    if (params[@"scaleResolutionDownBy"] != nil) {
+      [encoding setScaleResolutionDownBy: (NSNumber *) params[@"scaleResolutionDownBy"]];
+    }
+
+    return encoding;
+}
+
++ (RTCRtpTransceiverInit *) parseTransceiverOptions: (NSDictionary *) params {
+    RTCRtpTransceiverInit *transceiverInit = [RTCRtpTransceiverInit new];
+    
+    NSString *direction = [params objectForKey: @"direction"];
```

Comment:
Here we should check if the direction is not nil before proceeding, like we do on Android: https://github.com/react-native-webrtc/react-native-webrtc/blob/f2278cf36ddaff43b505f19cad558240a4997aab/android/src/main/java/com/oney/WebRTCModule/SerializeUtils.java#L292

Reply:
Just figured as the direction is set as Inactive by default, the same functionality occurs without checking if the direction is nil.

Should i still proceed?

Reply:
Ahhh wait alright, yes my thinking of that was backwards, will address that now 👍🏻